tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml

```diff
@@ -27,8 +27,8 @@ tests:
               Use this [link](https://github.com/microsoft/vscode/issues?page=3&q=is%3Aissue+is%3Aopen)."}
             version: HTTP/1.1
           output:
-            no_log_contains: |-
-              id "911100"|id "932200"|id "933210"|id "942200"|id "942260"|id "942370"|id "942430"|id "942440"
+            no_log_contains: |
+              id "911100" | id "932200" | id "933210" | id "941110" | id "942200" | id "942260" | id "942370" | id "942430" | id "942440"
```

Comment:
The log contains entries like `...[id "932200"]...`. The spaces in these expressions will prevent matches of any ID in the log.

Reply:
is there no way to ignore the whitespace? it looks a bit cleaner with the whitespace between each rule id.

Reply:
No. This is a regular expression. The new test format fixes this issue.

Reply:
Awesome, I'll push the fix in a sec.

Reply:
Other tests had whitespace between the rule IDs, and fixing those seems to have shown some regressions. 9508172-1 has it's content type set to JSON, but it's set to both urlencoded and JSON. Running the tests locally shows that the payloads aren't being sent correctly:
```
--a710eb21-A--
[04/Jul/2024:06:20:17.914344 +1000] ZoWygesbvPUTE8d6pGP3cAAAAE4 127.0.0.1 43160 127.0.0.1 80
--a710eb21-B--
POST /ocs/v2.php/apps/files_sharing/api/v1/shares HTTP/1.1
Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
Connection: close
Content-Length: 153
Content-Type: application/x-www-form-urlencoded, application/json
Host: localhost
User-Agent: OWASP CRS test agent

--a710eb21-C--
%7B%22path%22%3A%22%2Fpath%2Fto%2Ftmp%2Fexample%2F%22%2C%22shareType%22%3A3%2C%22password%22%3A%22%3Cscript%3E%22%2C%22attributes%22%3A%22%5B%5D%22%7D%0A

```
This doesn't look right. I see absolutely zero reason why some tests like 9508362-1 are failing.

Reply:
One thing that causes a couple of issues is are  a couple of lowercase `content-type` headers in the test definitions. Change these to `Content-Type` (go-ftw performs a case-insensitive check for the header and then adds adds it's own, if it couldn't find it).

Second, rule 9508316, for example, doesn't actually remove any targets or disable rules. So it's not surprising that rules are triggered. I also noticed that some tests, e.g., 9508316, still have spaces in the regular expression for `no_log_contains`.